src/ModelContextProtocol/AIContentExtensions.cs

```diff
@@ -62,9 +58,12 @@ public static AIContent ToAIContent(this ResourceContents content)
     {
         Throw.IfNull(content);
 
-        AIContent ac = content.Blob is not null && content.MimeType is not null ?
-            new DataContent(Convert.FromBase64String(content.Blob), content.MimeType) :
-            new TextContent(content.Text);
+        AIContent ac = content switch
+        {
+            BlobResourceContents blobResource => new DataContent(Convert.FromBase64String(blobResource.Blob), blobResource.MimeType ?? "application/octet-stream"),
+            TextResourceContents textResource => new TextContent(textResource.Text),
+            _ => throw new NotSupportedException($"Resource type '{content.GetType().Name}' is not supported.")
```

Comment:
Should we add an internal ctor to ResourceContents such that this case isn't actually possible?

Reply:
I've made it an abstract class, just to ensure that you can't create one which is "unknown", so that wouldn't be possible. But I'm unsure if raising errors like this is right from a server perspective.

Reply:
> I've made it an abstract class, just to ensure that you can't create one which is "unknown", so that wouldn't be possible.

Sure it is :smile:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABAzAAgIbADOALlHmMThgEw4BCehMAsAFADebO3OAlgHbEYUfngA29RjAAUAShzsAvm2Ws22HE3EwAJlVoARYbwBuunCElMFioA===

Not saying we need to, but it's definitely possible.

Reply:
Well yes, you can add a constructor but you can't call it... can you?

Reply:
The derived types in the same assembly can. That's the point. If nothing outside of the assembly can derive from it, and all of the types inside the assembly are in the switch, then the default case can never be hit. With the type abstract and its default protected ctor, someone outside of the assembly can still derive from it, and then you need that default case to handle the unexoected input.

Reply:
Ah cool, I hadn't considered that aspect